File: Remove Duplicates in-place from Sorted Array/code.py
def solve(arr, N):
    i = 0
    for j in range(1, N):
        if arr[i] != arr[j]:
            i += 1
            arr[i] = arr[j]
    print(arr[:i+1])
    

arr = [1,2,2,2,3]
N = len(arr)
solve(arr, N)

File: Remove Duplicates in-place from Sorted Array/code.cpp
#include <iostream>
#include <vector>
using namespace std;

void solve(vector<int>& arr, int N) {
    int i = 0;
    for (int j = 1; j < N; ++j) {
        if (arr[i] != arr[j]) {
            ++i;
            arr[i] = arr[j];
        }
    }
    for (int k = 0; k <= i; ++k) {
        cout << arr[k] << " ";
    }
}

int main() {
    vector<int> arr = {1, 2, 2, 2, 3};
    int N = arr.size();
    solve(arr, N);
    return 0;
}
